### PR TITLE
API

### PR DESCRIPTION
**Pull request by bartosz-kasprzyk**
I've added api data into every subpage in the movie browser. Please take a look at it and give your feedback! 😎

### CODE REVIEW

**Comment by przemyslaw-pacek** [comment]
It looks very good.

File: src/features/MovieList/index.js [hunk 1]
```diff
@@ -1,90 +1,49 @@
...
-                />
-            </Content>
-            <Pagination />
-        </Wrapper>
+        <>
+            {popularMovies.status === "loading" ? (
+                <Loading />
+            )
+                : popularMovies.status === "error" ? (
+                    "Error while fetching data from external API"
```
Comment (przemyslaw-pacek) [inline src/features/MovieList/index.js]: We have an Error component and it needs to be applied.

File: src/features/MovieList/usePopularMovies.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import { useState, useEffect } from "react";
+import { url_back, url_front_movie } from "../../common/API/requests";
+
+const page = "1";
+const url = `${url_front_movie}popular${url_back}&page=${page}`;
+
+export const usePopularMovies = () => {
+    const [popularMovies, setPopularMovies] = useState({
+        status: "loading",
+        data: [],
```
Comment (przemyslaw-pacek) [inline src/features/MovieList/usePopularMovies.js:11]: This element is unnecessary {data: [],).

File: src/features/MovieList/usePopularMovies.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    });
+
+    useEffect(() => {
+        const getPopularMovies = async () => {
+            try {
+                const response = await axios.get(url);
+                setPopularMovies({
+                    status: "success",
+                    data: response.data.results,
```
Comment (przemyslaw-pacek) [inline src/features/MovieList/usePopularMovies.js]: I would replace `data: response.data.results` with `data: response.data`. There were a few times when I needed other values ​​beyond "results".

File: src/features/MoviePage/index.js [hunk 4]
```diff
@@ -1,35 +1,64 @@
...
-            A young Chinese maiden disguises herself as a male warrior in order to save her father."
-            />
-            <Cast />
-            <Crew />
+            {movieDetails.status === "loading" ? (
+                <Loading />
+            )
+                : movieDetails.status === "error" ? (
+                    "Error while fetching data from external API"
+                )
```
Comment (przemyslaw-pacek) [inline src/features/MoviePage/index.js]: The same - <Error /> component.

File: src/features/MoviePage/useMovieCredits.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+                });
+            } catch (error) {
+                setMovieCredits({
+                    status: "error"
+                });
+                console.log(error);
+            }
+        };
+
+        setTimeout(getMovieCredits, 0);
```
Comment (przemyslaw-pacek) [inline src/features/MoviePage/useMovieCredits.js:32]: `setTimeout` is not needed if it has the value "0".

File: src/useGenres.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+                setGenres({
+                    data: response.data.genres,
+                });
+            } catch (error) {
+                console.log(error);
+            }
+        };
+
+        setTimeout(getGenres, 0);
+    }, []);
```
Comment (przemyslaw-pacek) [inline src/useGenres.js]: The same - setTimeout is not needed if it has the value "0".

**Review by przemyslaw-pacek — CHANGES_REQUESTED** [review]

File: src/common/API/requests.js
Comment (przemyslaw-pacek) [inline src/common/API/requests.js:1]: One api.js file is not enough?

File: src/features/MovieList/index.js (hunk 1 above)
Comment (bartosz-kasprzyk) [reply src/features/MovieList/index.js]: True! Totally forgot about it

File: src/features/MovieList/usePopularMovies.js (hunk 2 above)
Comment (bartosz-kasprzyk) [reply src/features/MovieList/usePopularMovies.js:11]: I think it is necessary, cause it doesn't always work without it and adding it resolved the errors.

File: src/useGenres.js (hunk 6 above)
Comment (bartek-kasprzyk) [reply src/useGenres.js:25]: I'm not sure if it is a bad practice, but deleting it causes issues, so I probably would've had to rewrite the function in a different way to solve it. Also I've read somewhere that setTimeout(0) schedules the callback after the shortest possible delay, so I think it works in this instance.

File: src/features/MoviePage/useMovieCredits.js (hunk 5 above)
Comment (bartosz-kasprzyk) [reply src/features/MoviePage/useMovieCredits.js:32]: I'm not sure if it is a bad practice, but deleting it causes issues, so I probably would've had to rewrite the function in a different way to solve it. Also I've read somewhere that setTimeout(0) schedules the callback after the shortest possible delay, so I think it works in this instance.

**Review by przemyslaw-pacek — APPROVED** [review]

**Review by VITALIKXXX — APPROVED** [review]

**Pull request merged**